lib/Runtime/Base/FunctionBody.cpp

```diff
@@ -1143,7 +1143,7 @@ namespace Js
         if (nestedCount > 0)
         {
             nestedArray = RecyclerNewPlusZ(m_scriptContext->GetRecycler(),
-                nestedCount*sizeof(FunctionProxy), NestedArray, nestedCount);
+                (nestedCount-1)*sizeof(FunctionProxy*), NestedArray, nestedCount);
```

Comment:
Have you tried? 
There are other example in the code base that have declares a 0 length array field.
e.g. TypePath's "assignment" field.